app/collections/collections-manager.jsx

```diff
@@ -1,46 +1,80 @@
 import React from 'react';
 import CollectionsCreateForm from './create-form';
 import CollectionSearch from './collection-search';
+import LoadingIndicator from '../components/loading-indicator';
 
 class CollectionsManager extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      adding: false,
       collections: [],
-      error: null,
-      hasCollectionSelected: false
+      errors: [],
+      hasCollectionSelected: true
     };
 
     this.addToCollections = this.addToCollections.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  onChange() {
+    const collections = this.search.getSelected();
+
+    if (collections.length > 0) {
+      this.setState({ hasCollectionSelected: false });
+    }
   }
 
   addToCollections() {
+    this.setState({ adding: true, errors: [] });
     const collections = this.search.getSelected();
-    if (!(collections.length > 0)) return;
+    if (collections.length === 0) return;
 
-    const promises = collections.map(collection =>
-      collection.collection.addLink('subjects', this.props.subjectIDs)
-    );
+    const promises = collections.map((searchResult) => {
+      const collection = searchResult.collection;
+      const subjectsToAdd = this.props.subjectIDs.filter((id) => {
+        if (!collection.links.subjects.includes(id)) return id;
+      });
+
+      if (subjectsToAdd.length === 0) {
+        const noSubjectsError = [
+          `There are no subjects to add or subjects are already added to ${collection.display_name}.`
+        ];
+        const errors = noSubjectsError.concat(this.state.errors);
+        this.setState({ errors });
+      }
+
+      return collection.addLink('subjects', subjectsToAdd)
+        .catch((error) => {
+          const newError = [`${collection.display_name}: ${error}`];
+          const errors = newError.concat(this.state.errors);
+          this.setState({ errors });
+        });
+    });
 
     Promise.all(promises)
       .then(() => {
-        this.props.onSuccess();
-      }).catch(error => this.setState({ error }));
+        if (this.state.errors.length === 0) this.props.onSuccess();
+        this.setState({ adding: false });
+      }).catch(error => this.setState({ adding: false, errors: [error.toString()] }));
   }
 
   render() {
     return (
       <div className="collections-manager">
-        <h1>Add Subject to Collection</h1>
+        <h1>Add Subject to Collection{' '}{this.state.adding && <LoadingIndicator />}</h1>
```

Comment:
This shouldn't be `h1` unless `Add Subject to Collection` is the top-level heading for the current page.